src/screens/MainScreen/index.js

```diff
@@ -1,18 +1,45 @@
-import React, { useEffect, useMemo } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { Text, View, TouchableOpacity, Image, Alert } from 'react-native';
 import MapView, { Marker } from 'react-native-maps';
 import { bool, string, array, func } from 'prop-types';
 import { connect } from 'react-redux';
+import Geolocation from '@react-native-community/geolocation';
 import styles from './styles';
 import { getTargets } from '../../actions';
 import { PROFILE_SCREEN } from '../../constants/screens';
 import targetIcon from '../../images/group2.png';
 import profileIcon from '../../images/page1.png';
+import pinIcon from '../../images/pinLocation.png';
 import { Spinner } from '../../components';
 
 const MainScreen = ({ getTargets, targets, loading, error }) => {
+  const [location, setLocation] = useState({});
+  const [region, setRegion] = useState({
+    latitude: 37.78825,
+    longitude: -122.4324,
+    latitudeDelta: 0.0922,
+    longitudeDelta: 0.0421
+  });
+
   useEffect(() => {
     getTargets();
+    Geolocation.getCurrentPosition(
+      position => {
+        const location = {
+          longitude: position.coords.longitude,
+          latitude: position.coords.latitude
+        };
+        setRegion({
+          latitude: location.latitude,
+          longitude: location.longitude,
+          latitudeDelta: 0.0922,
+          longitudeDelta: 0.0421
+        });
```

Comment:
```suggestion
        const { latitude, longitude } = location;
        setRegion({
          latitude,
          longitude,
          latitudeDelta: 0.0922,
          longitudeDelta: 0.0421
        });
```

Reply:
I'd destructure `position` and do this:
```suggestion
Geolocation.getCurrentPosition(
      ({ coords: { latitude, longitude } }) => {
        const location = {
          longitude,
          latitude
        };
        setRegion({
          latitude,
          longitude,
          latitudeDelta: 0.0922,
          longitudeDelta: 0.0421
        });
        .
        .
        .
```

Reply:
It there is a possibility that `coords` is not defined you can destructure like this: 
```suggestion
({ coords: { latitude, longitude } = {} })
```

Reply:
Cool, much better 👍